### PR TITLE
Add three ORL examples

### PR DESCRIPTION
**Pull request by annaluo676**
This is an on-going PR that will include three notebook example trying to solve operational research problems using reinforcement learning. Would like to gather some feedbacks along the way.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: reinforcement_learning/rl_bin_packing_newsvendor_vehicle_routing_ray_customEnv/rl_bin_packing_ray_custom.ipynb [hunk 1]
```diff
@@ -0,0 +1,588 @@
+{
```
Comment (bbalaji-ucsd) [inline reinforcement_learning/rl_bin_packing_newsvendor_vehicle_routing_ray_customEnv/rl_bin_packing_ray_custom.ipynb]: <blockquote>This notebook shows an example of how to use reinforcement learning technique to address an online and stochastic bin packing problem. In the classic version of bin packing, we are given items of different sizes and need to pack them into as few bins as possible. This problem is NP-hard. In the online stochastic version, items arrive one at a time and item sizes are drawn from an unknown distribution. The task is to find a feasible packing that minimizes the number of bins used to pack all of the items that arrive within the time horizon.</blockquote><p>This notebook shows an example of how to use reinforcement learning <strong>to solve the</strong> <strong>online stochastic bin packing</strong> problem. In the classic version of bin packing, we are given items of different sizes and need to pack them into as few bins as possible. In the online stochastic version, items arrive one at a time and item sizes are drawn from an unknown distribution. The task is to find a feasible packing that minimizes the number of bins used to pack all of the items that arrive within the time horizon.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/awslabs/amazon-sagemaker-examples/pull/934/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='0'/>

Comment (bbalaji-ucsd) [inline reinforcement_learning/rl_bin_packing_newsvendor_vehicle_routing_ray_customEnv/rl_bin_packing_ray_custom.ipynb]: <p>This is technical language, and difficult to understand. Can we add content from the slide deck, including pictures to explain the problem?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/awslabs/amazon-sagemaker-examples/pull/934/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='1'/>

Comment (bbalaji-ucsd) [inline reinforcement_learning/rl_bin_packing_newsvendor_vehicle_routing_ray_customEnv/rl_bin_packing_ray_custom.ipynb]: <p>training job run <strong>longer</strong></p><p> </p><blockquote>The sub-figures correspond to three categories of the item distribution, Perfectly Packable with Bounded Waste (BW), Perfectly Packable (PP) and Linear Waste (LW) correspondingly.&nbsp;</blockquote><p>The sub-figures correspond to three categories of the item distribution, Perfectly Packable with Bounded Waste (BW), Perfectly Packable (PP) and Linear Waste (LW) <strong>respectively</strong>.&nbsp;</p><p> </p><blockquote>You can try to replicate with a single machine with 4 GPUs and 32 CPUs for this experiment.</blockquote><p><strong>We use p3.8x instances in our experiments. </strong></p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/awslabs/amazon-sagemaker-examples/pull/934/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='32'/>

File: reinforcement_learning/rl_bin_packing_newsvendor_vehicle_routing_ray_customEnv/rl_news_vendor_ray_custom.ipynb [hunk 2]
```diff
@@ -0,0 +1,549 @@
+{
```
Comment (bbalaji-ucsd) [inline reinforcement_learning/rl_bin_packing_newsvendor_vehicle_routing_ray_customEnv/rl_news_vendor_ray_custom.ipynb]: <p>Remove the discrete version. It doesn't work. </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/awslabs/amazon-sagemaker-examples/pull/934/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='1'/>

File: reinforcement_learning/rl_resource_allocation_ray_customEnv/rl_bin_packing_ray_custom.ipynb [hunk 3]
```diff
@@ -0,0 +1,493 @@
+{
```
Comment (bbalaji-ucsd) [inline reinforcement_learning/rl_resource_allocation_ray_customEnv/rl_bin_packing_ray_custom.ipynb]: <blockquote>Assume each bin has size of&nbsp;10</blockquote><p>Assuming</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/awslabs/amazon-sagemaker-examples/pull/934/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='1'/>

Comment (bbalaji-ucsd) [inline reinforcement_learning/rl_resource_allocation_ray_customEnv/rl_bin_packing_ray_custom.ipynb]: <blockquote>Under the customized simulator, it is defined as:</blockquote><p>We define</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/awslabs/amazon-sagemaker-examples/pull/934/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='3'/>

File: reinforcement_learning/rl_resource_allocation_ray_customEnv/rl_vehicle_routing_problem_ray_custom.ipynb [hunk 4]
```diff
@@ -0,0 +1,456 @@
+{
```
Comment (bbalaji-ucsd) [inline reinforcement_learning/rl_resource_allocation_ray_customEnv/rl_vehicle_routing_problem_ray_custom.ipynb]: <blockquote>as it has a wide applicaiton in logistics</blockquote><p>application</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/awslabs/amazon-sagemaker-examples/pull/934/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='1'/>

Comment (bbalaji-ucsd) [inline reinforcement_learning/rl_resource_allocation_ray_customEnv/rl_vehicle_routing_problem_ray_custom.ipynb]: <blockquote>The city consists of mutually exclusive zones</blockquote><p>The city <strong>is a grid map and</strong> consists of mutually exclusive zones</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/awslabs/amazon-sagemaker-examples/pull/934/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='2'/>

**Comment by annaluo676** [comment]
Thanks, @bbalaji-ucsd. 
@yijiezz Can you please merge this PR?

**Pull request merged**